### ISSUE TITLE
Shutil rmtree failing on python 3.7

### ISSUE THREAD

**Issue by omangin** [jmcgeheeiv/pyfakefs#421]
The following test (adapted from [the documentation](http://jmcgeheeiv.github.io/pyfakefs/release/usage.html#patch-using-fake-filesystem-unittest)) raises an error on Python 3.7 although working on other versions.

```python
import os
import shutil

from pyfakefs.fake_filesystem_unittest import TestCase


class ExampleTestCase(TestCase):

    def setUp(self):
        self.setUpPyfakefs()

    def test_create_file(self):
        file_path = '/test/file.txt'
        self.assertFalse(os.path.exists(file_path))
        self.fs.CreateFile(file_path)
        self.assertTrue(os.path.exists(file_path))
        shutil.rmtree('/test/')
        self.assertFalse(os.path.exists(file_path))
```

```
E
======================================================================
ERROR: test_create_file (test.ExampleTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/test.py", line 17, in test_create_file
    shutil.rmtree('/test/')
  File "/tmp/env/lib/python3.7/shutil.py", line 485, in rmtree
    _rmtree_safe_fd(fd, path, onerror)
  File "/tmp/env/lib/python3.7/shutil.py", line 400, in _rmtree_safe_fd
    with os.scandir(topfd) as scandir_it:
  File "/tmp/e/lib/python3.7/site-packages/pyfakefs/fake_filesystem.py", line 3577, in scandir
    return self.filesystem.ScanDir(path)
  File "/tmp/e/lib/python3.7/site-packages/pyfakefs/fake_filesystem.py", line 2857, in ScanDir
    return self.ScanDirIter(self, path)
  File "/tmp/e/lib/python3.7/site-packages/pyfakefs/fake_filesystem.py", line 2813, in __init__
    self.path = self.filesystem.ResolvePath(path)
  File "/tmp/e/lib/python3.7/site-packages/pyfakefs/fake_filesystem.py", line 1738, in ResolvePath
    file_path = os.fspath(file_path)
TypeError: expected str, bytes or os.PathLike object, not int

----------------------------------------------------------------------
Ran 1 test in 0.002s

FAILED (errors=1)
```

**Comment by mrbean-bremen** [comment]
Thanks - I will have a look. We have some basic tests for `shutil` functionality, as we don't fake it and rely on it's usage of the faked functions, but we don't cover all of it - so with a changed implementation in a new Python version this may happen.

**Comment by mrbean-bremen** [comment]
This looks like a duplicate of #346, which was fixed in pyfakefs 3.4. This was due support for file descriptor path parameter in `os.scandir` added in Python 3.7 and used in `shutil.rmtree()`. 
Maybe you are using and older version of pyfakefs?

**Comment by omangin** [comment]
Indeed. This is fixed by version 3.4. Thanks and sorry for not having checked that.
